src/json.h

```diff
@@ -703,6 +706,71 @@ class JsonObject
 
         // useful debug info
         std::string line_number(); // for occasional use only
+
+        template <typename T>
```

Comment:
Did you forget to delete this code?
Long-term it's never going to be ok to put knowledge of any extensions we make to JSON in this module.  If you want to add hooks to let the caller provide various special handling, validation or tracking, that's fine, but it needs to have an arms-length API that keeps the changes outside of these files.